tests/datasource/fluent/integration/test_sql_datasources.py

```diff
@@ -710,5 +711,131 @@ def _raw_query_check_column_exists(
         return True
 
 
+@pytest.mark.parametrize(
+    "column_name",
+    [
+        # DDL: unquoted_lower_col ----------------------------------
+        param("unquoted_lower_col", id="str unquoted_lower_col"),
+        param('"unquoted_lower_col"', id='str "unquoted_lower_col"'),
+        param("UNQUOTED_LOWER_COL", id="str UNQUOTED_LOWER_COL"),
+        param('"UNQUOTED_LOWER_COL"', id='str "UNQUOTED_LOWER_COL"'),
+        # DDL: UNQUOTED_UPPER_COL ----------------------------------
+        param("unquoted_upper_col", id="str unquoted_upper_col"),
+        param('"unquoted_upper_col"', id='str "unquoted_upper_col"'),
+        param("UNQUOTED_UPPER_COL", id="str UNQUOTED_UPPER_COL"),
+        param('"UNQUOTED_UPPER_COL"', id='str "UNQUOTED_UPPER_COL"'),
+        # DDL: "quoted_lower_col"-----------------------------------
+        param("quoted_lower_col", id="str quoted_lower_col"),
+        param('"quoted_lower_col"', id='str "quoted_lower_col"'),
+        param("QUOTED_LOWER_COL", id="str QUOTED_LOWER_COL"),
+        param('"QUOTED_LOWER_COL"', id='str "QUOTED_LOWER_COL"'),
+        # DDL: "QUOTED_UPPER_COL" ----------------------------------
+        param("quoted_upper_col", id="str quoted_upper_col"),
+        param('"quoted_upper_col"', id='str "quoted_upper_col"'),
+        param("QUOTED_UPPER_COL", id="str QUOTED_UPPER_COL"),
+        param('"QUOTED_UPPER_COL"', id='str "QUOTED_UPPER_COL"'),
+    ],
+)
+class TestColumnIdentifiers:
```

Comment:
I don't believe this is an ID/PK test but I'm glad that this is back. Be had a lot of bugs around quoted identifiers and casing especially with snowflake.
If other tests in this file were deleted from this file, you may want to take another look to see if any others should come back. Happy to go through them with you.